### PR TITLE
fix: add in card creation per channel to uphold connection lifecycle …

### PR DESCRIPTION
**Pull request by jlbyrne**
…methods

Addresses https://github.com/brave-intl/creators-private-issues/issues/134

### CODE REVIEW

File: app/models/uphold_connection.rb [hunk 1]
```diff
@@ -260,7 +263,7 @@ def sync_connection!
         uphold_id: result.id,
         country: result.country
       )
-      create_uphold_cards if missing_card? && success
```
[the remark this answers is not in the document]
Comment (jlbyrne) [reply app/models/uphold_connection.rb:263]: The create uphold cards job checks to see if cards are already created.  It also exits early if there is an existing uphold connection for channel that has a card id.  No requests to uphold are made that won't need to be made.

[the remark this answers is not in the document]
Comment (tsmartt) [reply app/models/uphold_connection.rb:263]: I'd vote to keep this check in place so we don't queue unnecessary jobs

**Review by tsmartt — CHANGES_REQUESTED** [review]
Thanks for tackling this!

I made a small comment about the check. 

With these dependent destroy changes, what do you think about adding tests to ensure that upon disconnect, the old cards are destroyed?

**Comment by jlbyrne** [comment]
I think I added some for uphold when I made the other change, but I can add some for Gemini

**Review by tsmartt — APPROVED** [review]
LG2M. 

This historic code is a mess and I'll make a ticket to clean it up further. Notably, since there are 2 types of cards, something like 

```ruby
create_uphold_connection_card if missing_connection_card? && success
create_uphold_channel_cards if missing_channel_card? && success
```

can be done here.

**Pull request merged**